flair/training_utils.py

```diff
@@ -15,64 +17,94 @@ class Metric(object):
+
+    def precision(self, cls=None):
+        if self._tps[cls] + self._fps[cls] > 0:
+            return round(self._tps[cls] / (self._tps[cls] + self._fps[cls]), 4)
         return 0.0
 
-    def recall(self):
-        if self._tp + self._fn > 0:
-            return round(self._tp / (self._tp + self._fn), 4)
+    def recall(self, cls=None):
+        if self._tps[cls] + self._fns[cls] > 0:
+            return round(self._tps[cls] / (self._tps[cls] + self._fns[cls]), 4)
         return 0.0
 
-    def f_score(self):
-        if self.precision() + self.recall() > 0:
-            return round(2 * (self.precision() * self.recall()) / (self.precision() + self.recall()), 4)
+    def f_score(self, cls=None):
+        if self.precision(cls) + self.recall(cls) > 0:
+            return round(2 * (self.precision(cls) * self.recall(cls)) / (self.precision(cls) + self.recall(cls)), 4)
         return 0.0
 
-    def accuracy(self):
-        if self._tp + self._tn + self._fp + self._fn > 0:
-            return round((self._tp + self._tn) / (self._tp + self._tn + self._fp + self._fn), 4)
+    def accuracy(self, cls=None):
+        if self._tps[cls] + self._tns[cls] + self._fps[cls] + self._fns[cls] > 0:
+            return round(
+                (self._tps[cls] + self._tns[cls]) / (self._tps[cls] + self._tns[cls] + self._fps[cls] + self._fns[cls]),
+                4)
         return 0.0
 
     def to_tsv(self):
+        # gather all the classes
         return '{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}'.format(
-            self._tp, self._tn, self._fp, self._fn, self.precision(), self.recall(), self.f_score(), self.accuracy())
+            self.get_tp(), self.get_tn(), self.get_fp(), self.get_fn(), self.precision(), self.recall(), self.f_score(), self.accuracy())
 
     def print(self):
         log.info(self)
 
     @staticmethod
     def tsv_header(prefix=None):
         if prefix:
-            return '{0}_TP\t{0}_TN\t{0}_FP\t{0}_FN\t{0}_PRECISION\t{0}_RECALL\t{0}_F-SCORE\t{0}_ACCURACY'.format(prefix)
+            return 'CLS\t{0}_TP\t{0}_TN\t{0}_FP\t{0}_FN\t{0}_PRECISION\t{0}_RECALL\t{0}_F-SCORE\t{0}_ACCURACY'.format(
+                prefix)
 
-        return 'TP\tTN\tFP\tFN\tPRECISION\tRECALL\tF-SCORE\tACCURACY'
+        return 'CLS\TP\tTN\tFP\tFN\tPRECISION\tRECALL\tF-SCORE\tACCURACY'
 
     @staticmethod
     def to_empty_tsv():
         return '_\t_\t_\t_\t_\t_\t_\t_'
 
     def __str__(self):
-        return '{0:<10}\ttp: {1} - fp: {2} - fn: {3} - tn: {4} - precision: {5:.4f} - recall: {6:.4f} - accuracy: {7:.4f} - f1-score: {8:.4f}'.format(
-            self.name, self._tp, self._fp, self._fn, self._tn, self.precision(), self.recall(), self.accuracy(), self.f_score())
+        all_classes = self.get_classes()
+        all_lines = [
+            '{0:<10}\ttp: {1} - fp: {2} - fn: {3} - tn: {4} - precision: {5:.4f} - recall: {6:.4f} - accuracy: {7:.4f} - f1-score: {8:.4f}'.format(
+                MICRO_AVG_METRIC if cls == None else cls,
```

Comment:
I would take `self.name` if `cls is None` as the metric can be used in many different ways and it does not need to be the micro average of some classes.